analyzers/tests/SonarAnalyzer.UnitTest/Rules/Utilities/TokenTypeAnalyzerTest.Classifier.cs

```diff
@@ -811,7 +834,202 @@ public void M()
+            case iConst:
+                break;                
+        }
+        """)]
+    [DataRow("""
+        switch(AttributeTargets.Assembly)
+        {
+            case AttributeTargets.Class:
+                goto case AttributeTargets.[u:Enum];
+            case AttributeTargets.Enum:
+                break;                
+        }
+        """)]
+    [DataRow("""
+        goto [u:label];
+        label: ;
+        """)]
+    [DataRow("return [u:i];")]
+    [DataRow("throw [u:ex];")]
+    [DataRow("IEnumerable<int> YieldReturn() { yield return [u:i]; }")]
+    [DataRow("using(var x = [u:d]);")]
+    [DataRow("lock([u:d]);")]
+    [DataRow("try { } catch when ([u:b]) { }")]
+    public void IdentifierToken_SingleExpressionIdentifier(string statement, bool allowSemanticModel = false) =>
+        ClassifierTestHarness.AssertTokenTypes($$"""
+            using System;
+            using System.Collections.Generic;
+            using System.Linq;
+            using System.Threading.Tasks;
+
+            public class Test
+            {
+                public async Task<int> M()
+                {
+                    const int iConst = 0;
+                    var b = true;
+                    var i = 0;
+                    var ex = new Exception();
+                    var l = new List<Exception>();
+                    Task t = null;
+                    IDisposable d = null;
+                    {{statement}}
+                    return 0;
+                }
+            }
+            """, allowSemanticModel);
+
+#if NET
+
+    [DataTestMethod]
+    [DataRow("_ = [u:r] with { [u:A] = [u:iConst] };", false)]
+    [DataRow("_ = [u:r] is ([u:iConst], [t:Int32]);", true)] // semantic model must be called for iConst and Int32
+    [DataRow("_ = [u:l][^[u:iConst]];", false)]
+    [DataRow("_ = [u:a][[u:iConst]..^([u:iConst]-1)];", false)]
+    [DataRow("foreach((var [u:x], int [u:y]) in ts);", false)]
+    [DataRow("foreach(var ([u:x], [u:y]) in ts);", false)]
+    public void IdentifierToken_SingleExpressionIdentifier_NetCore(string statement, bool allowSemanticModel) =>
+        ClassifierTestHarness.AssertTokenTypes($$"""
+            using System;
+            using System.Collections.Generic;
+            using System.Linq;
+            using System.Threading.Tasks;
+
+            public record R(int A, int B);
+
+            public class Test
+            {
+                public async Task M()
+                {
+                    const int iConst = 0;
+                    var l = new List<Exception>();
+                    var ts = new (int, int)[0];
+                    var a = new int[0];
+                    var r = new R(1, 1);
+                    {{statement}}
+                }
+            }
+            """, allowSemanticModel);
+
+#endif
+
+    [DataTestMethod]
+    [DataRow("[Obsolete([u:sConst])]")]
+    [DataRow("[AttributeUsage(AttributeTargets.All, AllowMultiple = [u:bConst])]")]
```

Comment:
You can add also:
`[DataRow("[AttributeUsage([t:AttributeTargets].All, AllowMultiple = [u:bConst])]", true)]` for completeness.